### PR TITLE
Fix issues hit against debian8.11 + hyperv platform.

### PR DESCRIPTION
**Pull request by LiliDeng**
Fix issues we hit during run msft + debian8.11 on hyperv.

### CODE REVIEW

**Comment by LiliDeng** [comment]
Tests results against Debian8.11 on hyperv.
```
[LISAv2 Test Results Summary]
Test Run On           : 08/02/2019 04:08:16
VHD Under Test        : debian_8.11.vhdx
Initial Kernel Version: 3.16.0-6-amd64
Final Kernel Version  : 4.19.60-adc98d77678b
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 4.22 

[LISAv2 Test Results Summary]
Test Run On           : 08/02/2019 06:06:42
VHD Under Test        : debian_8.11.vhdx
Initial Kernel Version: 3.16.0-6-amd64
Final Kernel Version  : 4.19.60-adc98d77678b
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:12

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 DYNAMIC_MEMORY       DYNAMIC-MEMORY-HOT-ADD                                                            PASS                 9.35 
```

**Comment by LiliDeng** [comment]
Regression tests against ubuntu 18.04 on hyperv.
```
[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:29:26
VHD Under Test        : Ubuntu_18.04.1.vhdx
Initial Kernel Version: 4.15.0-38-generic
Final Kernel Version  : 4.19.60-adc98d77678b
Total Test Cases      : 3 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:38

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 DYNAMIC_MEMORY       DYNAMIC-MEMORY-HOT-ADD                                                            PASS                10.16 
    2 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 5.12 
    3 NETWORK              ETHTOOL-CHECK-STATISTICS                                                          PASS                12.07 
```

**Comment by LiliDeng** [comment]
NET-IFUP-IFDOWN is also supported on azure. Regression tests for test case NET-IFUP-IFDOWN on azure platform.
```
[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:13:10
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Initial Kernel Version: 4.18.0-1024-azure
Final Kernel Version  : 4.18.0-1024-azure
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:8

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 5.03 

[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:57:58
ARM Image Under Test  : credativ : Debian : 9-backports : Latest
Initial Kernel Version: 4.19.0-0.bpo.5-cloud-amd64
Final Kernel Version  : 4.19.0-0.bpo.5-cloud-amd64
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 4.98 

[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:58:11
ARM Image Under Test  : RedHat : RHEL : 7.5 : Latest
Initial Kernel Version: 3.10.0-862.34.2.el7.x86_64
Final Kernel Version  : 3.10.0-862.34.2.el7.x86_64
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 5.05 

[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:58:23
ARM Image Under Test  : OpenLogic : CentOS : 7.5 : Latest
Initial Kernel Version: 3.10.0-862.11.6.el7.x86_64
Final Kernel Version  : 3.10.0-862.11.6.el7.x86_64
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:8

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 5.21 

[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:58:45
ARM Image Under Test  : Oracle : Oracle-Linux : 7.6 : Latest
Initial Kernel Version: 4.14.35-1844.1.3.el7uek.x86_64
Final Kernel Version  : 4.14.35-1844.1.3.el7uek.x86_64
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:8

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                  5.3 


[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:58:58
ARM Image Under Test  : SUSE : SLES : 15 : Latest
Initial Kernel Version: 4.12.14-5.30-azure
Final Kernel Version  : 4.12.14-5.30-azure
Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
Total Time (dd:hh:mm) : 0:0:5

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                SKIPPED                 0.92 


[LISAv2 Test Results Summary]
Test Run On           : 08/05/2019 10:59:11
ARM Image Under Test  : CoreOS : CoreOS : Stable : Latest
Initial Kernel Version: 4.19.56-coreos-r1
Final Kernel Version  : 4.19.56-coreos-r1
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 5.29 
```

File: Testscripts/Linux/NET-IFUP-IFDOWN.sh [hunk 1]
```diff
@@ -89,19 +89,32 @@ done
...
 TestCount=1
 ChangeInterfaceState "up"
-kill "$(pidof dhclient)"
-dhclient -r && dhclient
+ipAddress=$(ip addr show eth0 | grep "inet\b")
+if [ -z "$ipAddress" ]; then
+    kill "$(pidof dhclient)"
+    dhclient -r && dhclient
+fi
 sleep 15
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/NET-IFUP-IFDOWN.sh]: if there is no dhclient service starting, can you move sleep statement inside of if statement?

File: Testscripts/Linux/NET-IFUP-IFDOWN.sh [hunk 2]
```diff
@@ -89,19 +89,32 @@ done
...
+    if [ $? -ne 0 ]; then
+        LogMsg "Run ip route add $default_route"
+        ip route add $default_route
+    fi
+    ipAddress=$(ip addr show eth0 | grep "inet\b")
+    if [ -z "$ipAddress" ]; then
+        kill "$(pidof dhclient)"
+        dhclient -r && dhclient
+    fi
     sleep 15
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/NET-IFUP-IFDOWN.sh]: Also please move sleep statement to inside of if statement.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/NET-IFUP-IFDOWN.sh [hunk 3]
```diff
@@ -89,20 +89,34 @@ done
...
-    dhclient -r && dhclient
-    sleep 15
+    ip route show | grep default
+    # Add default route when miss it after run ip link down/up
+    if [ $? -ne 0 ]; then
+        LogMsg "Run ip route add $default_route"
+        ip route add $default_route
+    fi
+    ipAddress=$(ip addr show eth0 | grep "inet\b")
+    if [ -z "$ipAddress" ]; then
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/NET-IFUP-IFDOWN.sh:115]: No need to check if it recovers?

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/NET-IFUP-IFDOWN.sh (hunk 3 above)
Comment (LiliDeng) [reply Testscripts/Linux/NET-IFUP-IFDOWN.sh:115]: > No need to check if it recovers?

Some distro will lose the IP if run network down and up, then run dhclient to recover it.
Some distro won't lose ip, then no need run dhclient.
So add check here.

**Review by xian123 — APPROVED** [review]

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**